crates/aggchain-proof-builder/src/lib.rs

```diff
@@ -71,11 +71,9 @@ pub struct AggchainProofBuilderResponse {
 #[derive(Clone)]
 #[allow(unused)]
 pub struct AggchainProofBuilder {
-    /// Mainnet node rpc client.
-    l1_client: Arc<AlloyProvider>,
-
-    /// L2 node rpc client.
-    l2_client: Arc<AlloyProvider>,
+    /// Client for interacting with the smart contracts relevant for the
+    /// aggchain prover.
+    contracts_client: Arc<AggchainProofContractsClient>,
```

Comment:
You can use a generic for further test capabilities

Reply:
Will do, I'll refactor for https://github.com/agglayer/provers/issues/63